### PR TITLE
Feature/vdx 149/profiles enum

### PR DESCRIPTION
**Pull request by hrehman-sphereon**

### CODE REVIEW

**Review by sksadjad — CHANGES_REQUESTED** [review]
overall looks good. two small notes, `profile` seems a bit off, because we're talking about different versions of the SIOP protocol here. and two remove two of the three methods you've added: `withProfileStr`, `addProfile`, `withProfiles`. and make the remaining one accept both list and singular value

**Comment by hrehman-sphereon** [comment]
```

public withProfiles(profiles: string | Profile | Array<Profile>): OPBuilder {
    if(typeof profiles === "string") {
      this.withProfileStr(profiles);
    } else if (profiles instanceof Profile) {
      this.addProfile(profiles as unknown as Profile);
    } else if (Array.isArray(profiles)) {
      this.withProfilesList(profiles);
    }

    return this;
  }
```
  
  `profiles instanceof Profile` .... actually have to be implemented like so ... `Array.isArray(profiles)`
  
  
  https://stackoverflow.com/questions/53672528/using-instanceof-with-typescript-union-types-customobject-customobject
  
  
This does not seem an elegant solution.

**Comment by hrehman-sphereon** [comment]
**This document is not a specification, but a profile. It outlines existing specifications required for implementations to interoperate among each other.**

Quoted from : https://identity.foundation/jwt-vc-presentation-profile/

I think the word "profile" does work... do you have a better word in mind ?

File: src/main/types/SupportedVersion.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export enum SupportedVersion {
```
Comment (sksadjad) [inline src/main/types/SupportedVersion.ts:1]: add this to our SIOP types

File: test/OPBuilder.spec.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { IVerifyCallbackArgs, IVerifyCredentialResult } from '@sphereon/wellknown-dids-client';
```
Comment (sksadjad) [inline test/OPBuilder.spec.ts:1]: we have a couple of builder tests already in the `OP.request.spec.ts` please move these to that test suite

File: test/RPBuilder.spec.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import { IVerifyCallbackArgs, IVerifyCredentialResult } from '@sphereon/wellknown-dids-client';
```
Comment (sksadjad) [inline test/RPBuilder.spec.ts:1]: we have a couple of builder tests already in the `RP.request.spec.ts` please move these to that test suite

**Review by sksadjad — CHANGES_REQUESTED** [review]
getting there, some small changes requested

File: src/main/types/SupportedVersion.ts (hunk 1 above)
Comment (hrehman-sphereon) [reply src/main/types/SupportedVersion.ts]: fixed

File: test/OPBuilder.spec.ts (hunk 2 above)
Comment (hrehman-sphereon) [reply test/OPBuilder.spec.ts]: fixed

File: test/RPBuilder.spec.ts (hunk 3 above)
Comment (hrehman-sphereon) [reply test/RPBuilder.spec.ts]: fixed

**Review by sksadjad — APPROVED** [review]
looks good to me

**Pull request merged**